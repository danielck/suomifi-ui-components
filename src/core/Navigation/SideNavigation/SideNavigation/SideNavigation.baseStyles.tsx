import { font } from '../../../theme/reset';
import { css } from 'styled-components';
import { SuomifiTheme } from '../../../theme';

export const baseStyles = (theme: SuomifiTheme) => css`
  ${font(theme)('bodyText')}
  &.fi-side-navigation {
    &--small-screen {
      border: 1px solid ${theme.colors.depthLight1};
      .fi-side-navigation_heading {
        ${font(theme)('heading4')}
        width: 100%;
        padding: ${theme.spacing.m};

        &:focus {
          position: relative;
          outline: 0;

          &:after {
            ${theme.focus.absoluteFocus}
          }
        }

        .fi-icon {
          height: 25px;
          width: 25px;
        }
      }

      .fi-side-navigation-item {
        &:last-child {
          &:after {
            display: none;
          }
        }
      }
    }

    .fi-side-navigation_heading {
      ${font(theme)('heading4')}
      display: flex;
      justify-content: space-between;
      align-items: center;
      padding: ${theme.spacing.xs} ${theme.spacing.s};

      .fi-side-navigation_heading_inner {
        display: flex;
        align-items: center;
      }

      .fi-static-icon {
        height: 45px;
        width: 45px;
        margin-right: ${theme.spacing.s};
      }
    }

    .fi-side-navigation_list {
      list-style-type: none;
      margin: 0;
      padding: 0;
    }
  }
`;
