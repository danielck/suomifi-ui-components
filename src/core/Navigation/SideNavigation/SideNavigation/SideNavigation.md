### Basic usage

Please use `<RouterLink>` as the child of `<SideNavigationItem>` to get inteded CSS styles. `<RouterLink>` is polymorphic, and can be rendered as any component of your choice, for example React Router Link.

```js
import {
  SideNavigation,
  SideNavigationItem,
  RouterLink,
  ExternalLink
} from 'suomifi-ui-components';

const CustomButton = (props) => {
  const { children, ...passProps } = props;
  return <button {...passProps}>{props.children}</button>;
};

<div style={{ width: '350px' }}>
  <SideNavigation
    heading="Economy"
    icon="piggyBank"
    aria-label="Main"
  >
    <SideNavigationItem
      subLevel={1}
      content={
        <RouterLink href="/" aria-current="location">
          Personal economy
        </RouterLink>
      }
    >
      <SideNavigationItem
        subLevel={2}
        content={
          <RouterLink href="/" aria-current="location">
            Crisis situations in personal finances
          </RouterLink>
        }
      >
        <SideNavigationItem
          subLevel={3}
          content={
            <RouterLink href="/">
              If you are unable to pay your debts
            </RouterLink>
          }
        />
        <SideNavigationItem
          subLevel={3}
          selected
          content={
            <RouterLink href="/" aria-current="page">
              Advice on banking and insurance matters
            </RouterLink>
          }
        />
      </SideNavigationItem>
      <SideNavigationItem
        subLevel={2}
        content={
          <RouterLink href="/">Last will and testament</RouterLink>
        }
      />
    </SideNavigationItem>
    <SideNavigationItem
      subLevel={1}
      content={
        <RouterLink
          asComponent={ExternalLink}
          href="https://suomi.fi"
          hideIcon
        >
          Taxation and public economy
        </RouterLink>
      }
    />
    <SideNavigationItem
      content={
        <RouterLink
          asComponent={CustomButton}
          onClick={() => console.log('Nav item clicked')}
        >
          Consumer protection
        </RouterLink>
      }
      subLevel={1}
    />
    <SideNavigationItem
      disabled
      content={
        <RouterLink aria-disabled role="link">
          Finance
        </RouterLink>
      }
      subLevel={1}
    />
  </SideNavigation>
</div>;
```

For frameworks where its internal link component is used as a wrapper for the actual link, for example NextJS, the following approach can be used:

```jsx static
<SideNavigationItem
  content={
    <NextJSLink href="/some-route" passHref>
      <RouterLink>Economy</RouterLink>
    </NextJSLink>
  }
/>
```

### Small screen variant

```js
import {
  SideNavigation,
  SideNavigationItem,
  RouterLink,
  ExternalLink
} from 'suomifi-ui-components';

const CustomButton = (props) => {
  const { children, ...passProps } = props;
  return <button {...passProps}>{props.children}</button>;
};

<div style={{ width: '350px' }}>
  <SideNavigation
    variant="smallScreen"
    heading="Economy"
    icon="piggyBank"
    initiallyExpanded={false}
    aria-label="Main"
  >
    <SideNavigationItem
      subLevel={1}
      content={
        <RouterLink href="/" aria-current="location">
          Personal economy
        </RouterLink>
      }
    >
      <SideNavigationItem
        subLevel={2}
        content={
          <RouterLink href="/" aria-current="location">
            Crisis situations in personal finances
          </RouterLink>
        }
      >
        <SideNavigationItem
          subLevel={3}
          content={
            <RouterLink href="/">
              If you are unable to pay your debts
            </RouterLink>
          }
        />
        <SideNavigationItem
          subLevel={3}
          selected
          content={
            <RouterLink href="/" aria-current="page">
              Advice on banking and insurance matters
            </RouterLink>
          }
        />
      </SideNavigationItem>
      <SideNavigationItem
        subLevel={2}
        content={
          <RouterLink href="/">Last will and testament</RouterLink>
        }
      />
    </SideNavigationItem>
    <SideNavigationItem
      subLevel={1}
      content={
        <RouterLink
          asComponent={ExternalLink}
          href="https://suomi.fi"
          hideIcon
        >
          Taxation and public economy
        </RouterLink>
      }
    />
    <SideNavigationItem
      content={
        <RouterLink
          asComponent={CustomButton}
          onClick={() => console.log('Nav item clicked')}
        >
          Consumer protection
        </RouterLink>
      }
      subLevel={1}
    />
    <SideNavigationItem
      disabled
      content={
        <RouterLink aria-disabled role="link">
          Finance
        </RouterLink>
      }
      subLevel={1}
    />
  </SideNavigation>
</div>;
```
