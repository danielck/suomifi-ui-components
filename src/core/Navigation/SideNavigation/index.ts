export {
  SideNavigation,
  SideNavigationProps,
} from './SideNavigation/SideNavigation';
export {
  SideNavigationItem,
  SideNavigationItemProps,
} from './SideNavigationItem/SideNavigationItem';
