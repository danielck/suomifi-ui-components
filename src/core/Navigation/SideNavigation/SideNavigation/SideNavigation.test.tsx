/* import React from 'react';
import { render } from '@testing-library/react';
import { axeTest } from '../../../../utils/test';

import { SideNavigation } from './SideNavigation';
import { SideNavigationItem } from '../SideNavigationItem/SideNavigationItem';
import { ExternalLink, RouterLink } from '../../../Link';
import { StaticChip } from '../../../Chip'; 

interface TestProps {
  children: string;
  onClick?: () => void;
  role?: string;
  tabIndex?: number;
}

const Comp = ({ children, ...passProps }: TestProps) => (
  <div {...passProps}>{children}</div>
);

const TestSideNavigation = (
  <div data-testid="test-paragraph">
    <SideNavigation>
      <SideNavigationItem aria-label="16 unread messages">
        <RouterLink
          asComponent={Comp}
          onClick={() => console.log('Nav item clicked')}
          role="button"
          tabIndex={0}
        >
          Inbox
          <StaticChip style={{ marginLeft: '15px' }}>16</StaticChip>
        </RouterLink>
      </SideNavigationItem>
      <SideNavigationItem>
        <RouterLink asComponent={ExternalLink} href="https://suomi.fi" hideIcon>
          Sent
        </RouterLink>
      </SideNavigationItem>
      <SideNavigationItem selected ariaCurrent="page">
        <RouterLink
          asComponent={ExternalLink}
          href="https://www.suomi.fi"
          hideIcon
        >
          New Message
        </RouterLink>
      </SideNavigationItem>
      <SideNavigationItem>
        <RouterLink
          asComponent={ExternalLink}
          href="https://www.suomi.fi"
          hideIcon
        >
          Drafts
        </RouterLink>
      </SideNavigationItem>
      <SideNavigationItem>
        <RouterLink
          asComponent={ExternalLink}
          href="https://www.suomi.fi"
          hideIcon
        >
          Settings
        </RouterLink>
      </SideNavigationItem>
      <SideNavigationItem>
        <RouterLink>Devices</RouterLink>
      </SideNavigationItem>
    </SideNavigation>
  </div>
);

test('calling render with the same component on the same container does not remount', () => {
  const ParagraphRendered = render(TestSideNavigation);
  const { container } = ParagraphRendered;
  expect(container.firstChild).toMatchSnapshot();
});

test('should not have basic accessibility issues', axeTest(TestSideNavigation)); */
