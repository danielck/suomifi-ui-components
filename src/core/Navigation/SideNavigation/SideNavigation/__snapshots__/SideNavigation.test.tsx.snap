// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling render with the same component on the same container does not remount 1`] = `
.c8 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  background-color: transparent;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8::before,
.c8::after {
  box-sizing: border-box;
}

.c8:hover,
.c8:active,
.c8:focus,
.c8:focus-within {
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c0::before,
.c0::after {
  box-sizing: border-box;
}

.c6 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: list-item;
}

.c6::before,
.c6::after {
  box-sizing: border-box;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
}

.c4::before,
.c4::after {
  box-sizing: border-box;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c2::before,
.c2::after {
  box-sizing: border-box;
}

.c5 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  list-style-type: decimal;
  margin-top: 1em;
  margin-bottom: 1em;
  margin-left: 0;
  margin-right: 0;
  padding-left: 40px;
}

.c5::before,
.c5::after {
  box-sizing: border-box;
}

.c10 {
  vertical-align: baseline;
}

.c10.fi-icon {
  display: inline-block;
}

.c10 .fi-icon-base-fill {
  fill: currentColor;
}

.c10 .fi-icon-base-stroke {
  stroke: currentColor;
}

.c10.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c10.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c3 {
  vertical-align: baseline;
}

.c3.fi-static-icon {
  display: inline-block;
}

.c3.fi-icon--cursor-pointer {
  cursor: pointer;
}

.c3.fi-icon--cursor-pointer * {
  cursor: inherit;
}

.c1 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c1.fi-side-navigation--small-screen {
  border: 1px solid hsl(202,7%,80%);
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_heading {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
  width: 100%;
  padding: 20px;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_heading:focus {
  position: relative;
  outline: 0;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_heading:focus:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation_heading .fi-icon {
  height: 25px;
  width: 25px;
}

.c1.fi-side-navigation--small-screen .fi-side-navigation-item:last-child:after {
  display: none;
}

.c1.fi-side-navigation .fi-side-navigation_heading {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 24px;
  font-weight: 600;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 15px;
}

.c1.fi-side-navigation .fi-side-navigation_heading .fi-side-navigation_heading_inner {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1.fi-side-navigation .fi-side-navigation_heading .fi-static-icon {
  height: 45px;
  width: 45px;
  margin-right: 15px;
}

.c1.fi-side-navigation .fi-side-navigation_list {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c7 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 600;
}

.c7.fi-side-navigation-item {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
}

.c7.fi-side-navigation-item:after {
  content: '';
  position: absolute;
  bottom: -1px;
  left: 2.5%;
  width: 95%;
  height: 1px;
  background: hsl(215,100%,97%);
}

.c7.fi-side-navigation-item:hover:after,
.c7.fi-side-navigation-item:active:after {
  background: transparent;
}

.c7.fi-side-navigation-item:focus,
.c7.fi-side-navigation-item:focus-within {
  outline: 0;
  box-shadow: none;
  border: none;
}

.c7.fi-side-navigation-item .fi-link--router {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px 20px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
  border: none;
  color: hsl(212,63%,45%);
  background: transparent;
}

.c7.fi-side-navigation-item .fi-link--router:visited {
  color: hsl(212,63%,45%);
}

.c7.fi-side-navigation-item .fi-link--router:hover,
.c7.fi-side-navigation-item .fi-link--router:active {
  background: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item .fi-link--router:focus {
  position: relative;
  outline: 0;
  box-shadow: none;
  border: none;
}

.c7.fi-side-navigation-item .fi-link--router:focus:before {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
  z-index: 9999;
}

.c7.fi-side-navigation-item.fi-side-navigation-item--child-selected > span > .fi-link--router {
  font-weight: 600;
}

.c7.fi-side-navigation-item .fi-side-navigation-item_sub-list {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c7.fi-side-navigation-item .fi-side-navigation-item_content-wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  position: relative;
}

.c7.fi-side-navigation-item .fi-side-navigation-item_content-wrapper .fi-icon {
  width: 25px;
  height: 25px;
  pointer-events: none;
  position: absolute;
  top: 11px;
  right: 15px;
  color: hsl(212,63%,45%);
}

.c7.fi-side-navigation-item--level-1.fi-side-navigation-item--selected {
  background: hsl(212,63%,45%);
  border-left: 4px solid hsl(212,63%,45%);
}

.c7.fi-side-navigation-item--level-1.fi-side-navigation-item--selected > span > .fi-link--router {
  text-transform: uppercase;
  color: hsl(0,0%,100%);
  font-weight: 600;
}

.c7.fi-side-navigation-item--level-1.fi-side-navigation-item--selected > span > .fi-link--router:hover,
.c7.fi-side-navigation-item--level-1.fi-side-navigation-item--selected > span > .fi-link--router:active {
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-1.fi-side-navigation-item--child-selected {
  border-left: 4px solid hsl(212,63%,45%);
}

.c7.fi-side-navigation-item--level-1 > span > .fi-link--router {
  text-transform: uppercase;
}

.c7.fi-side-navigation-item--level-2.fi-side-navigation-item--selected > span > .fi-link--router {
  font-weight: 600;
  background: hsl(212,63%,45%);
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-2.fi-side-navigation-item--selected > span > .fi-link--router + .fi-icon {
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-2 > span > .fi-link--router {
  text-transform: none;
  padding-left: 30px;
  padding-right: 60px;
  color: hsl(212,63%,45%);
  background: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-2 > span > .fi-link--router:hover + .fi-icon,
.c7.fi-side-navigation-item--level-2 > span > .fi-link--router:active + .fi-icon {
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-2 > .fi-side-navigation-item_content-wrapper:hover {
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-3.fi-side-navigation-item--selected > span > .fi-link--router {
  font-weight: 600;
  background: hsl(212,63%,45%);
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-3:hover,
.c7.fi-side-navigation-item--level-3:active {
  color: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--level-3 > span > .fi-link--router {
  text-transform: none;
  padding-left: 50px;
  color: hsl(212,63%,45%);
  background: hsl(212,63%,98%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c7.fi-side-navigation-item--disabled:hover,
.c7.fi-side-navigation-item--disabled:active {
  background: hsl(0,0%,100%);
}

.c7.fi-side-navigation-item--disabled .fi-link--router {
  color: hsl(202,7%,67%);
  cursor: not-allowed;
}

.c7.fi-side-navigation-item--disabled .fi-link--router:hover,
.c7.fi-side-navigation-item--disabled .fi-link--router:active,
.c7.fi-side-navigation-item--disabled .fi-link--router:visited {
  background: transparent;
  color: hsl(202,7%,67%);
}

.c12 {
  position: absolute;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
}

.c11 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c11.fi-link {
  color: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11.fi-link:hover,
.c11.fi-link:active,
.c11.fi-link:focus,
.c11.fi-link:focus-within {
  color: hsl(212,63%,45%);
}

.c11.fi-link:focus,
.c11.fi-link:focus-within {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11.fi-link:focus {
  outline: 0;
  border-radius: 2px;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
}

.c11.fi-link:hover,
.c11.fi-link:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11.fi-link:visited {
  color: hsl(284,36%,45%);
}

.c11 .fi-link_icon {
  padding-left: 4px;
  font-size: 16px;
  box-sizing: content-box;
}

.c9 {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
}

.c9.fi-link--router {
  color: hsl(212,63%,45%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9.fi-link--router:hover,
.c9.fi-link--router:active,
.c9.fi-link--router:focus,
.c9.fi-link--router:focus-within {
  color: hsl(212,63%,45%);
}

.c9.fi-link--router:focus,
.c9.fi-link--router:focus-within {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9.fi-link--router:focus {
  outline: 0;
  border-radius: 2px;
  box-shadow: 0 0 0 2px hsl(196,77%,44%);
}

.c9.fi-link--router:hover,
.c9.fi-link--router:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9.fi-link--router:visited {
  color: hsl(284,36%,45%);
}

<div
  class="c0 fi-side-navigation c1"
>
  <div
    class="c0 fi-side-navigation_heading"
  >
    <span
      class="c2 fi-side-navigation_heading_inner"
    >
      <svg
        aria-hidden="true"
        class="c3 fi-icon fi-static-icon"
        focusable="false"
        height="1em"
        viewBox="0 0 80 80"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            class="fi-icon-illustrative-base-fill"
            d="M23.611 62.508c-4.253-2.836-7.23-6.604-8.611-10.896L16.904 51c1.24 3.857 3.944 7.261 7.817 9.844l-1.11 1.664M40.255 68.011c-2.825 0-5.603-.363-8.255-1.08l.522-1.93c4.968 1.342 10.509 1.34 15.472-.001l.522 1.93a31.68 31.68 0 01-8.261 1.081M57.07 63.058l-1.11-1.664c5.437-3.625 8.555-8.923 8.555-14.535 0-5.991-3.468-11.511-9.515-15.145L56.03 30c6.664 4.005 10.485 10.15 10.485 16.859 0 6.292-3.442 12.196-9.445 16.199"
            fill="#A5ADB1"
          />
          <path
            class="fi-icon-illustrative-base-fill"
            d="M19 52.943h-8c-1.654 0-3-1.346-3-3v-6c0-1.654 1.346-3 3-3h3.296c1.081-2.761 3.332-5.604 6.704-8.466V22l8.14 5.814c2.585-1.015 4.86-1.871 7.86-1.871v2c-2.723 0-4.783.81-7.393 1.836l-.747.293L23 25.886v7.527l-.361.3c-3.516 2.916-5.827 5.868-6.687 8.537l-.223.693H11c-.552 0-1 .449-1 1v6c0 .551.448 1 1 1h8v2M33 72H23V58h2v12h6v-7h2v9M57 72H47v-9h2v7h6V58h2v14"
            fill="#A5ADB1"
          />
          <path
            class="fi-icon-illustrative-base-fill"
            d="M26 45c-2.757 0-5-2.243-5-5h2c0 1.654 1.346 3 3 3s3-1.346 3-3h2c0 2.757-2.243 5-5 5M65 48v-2c2.757 0 5-2.243 5-5h2c0 3.86-3.141 7-7 7"
            fill="#A5ADB1"
          />
          <path
            class="fi-icon-illustrative-highlight-fill"
            d="M49.601 32.8l-1.202-1.6A3.966 3.966 0 0050 28c0-2.206-1.794-4-4-4s-4 1.794-4 4c0 .875.277 1.705.801 2.402l-1.598 1.202A5.948 5.948 0 0140 28c0-3.309 2.691-6 6-6s6 2.691 6 6a5.956 5.956 0 01-2.399 4.8M47 14h2V8h-2z"
            fill="#E97025"
          />
          <path
            class="fi-icon-illustrative-base-fill"
            d="M52.4 35.8C48.99 33.243 44.28 32 38 32v-2c6.729 0 11.832 1.374 15.6 4.2l-1.2 1.6"
            fill="#A5ADB1"
          />
          <path
            class="fi-icon-illustrative-highlight-fill"
            d="M43 20h2v-6h-2zM47 18h2v-2h-2zM43 12h2v-2h-2z"
            fill="#E97025"
          />
          <path
            class="fi-icon-illustrative-base-fill"
            d="M39 60h2v-2h-2zM43 60h2v-2h-2zM35 60h2v-2h-2zM8 72h64v-2H8z"
            fill="#A5ADB1"
          />
          <path
            d="M0 0h80v80H0z"
          />
        </g>
      </svg>
      Economy
    </span>
  </div>
  <nav
    aria-label="Test"
    class="c4"
  >
    <ul
      class="c5 fi-side-navigation_list"
    >
      <li
        class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-1 fi-side-navigation-item--child-selected"
      >
        <span
          class="c2 fi-side-navigation-item_content-wrapper"
        >
          <a
            aria-current="location"
            class="c8 fi-link--router c9"
            href="/"
          >
            Personal economy
          </a>
        </span>
        <ul
          class="c5 fi-side-navigation-item_sub-list"
        >
          <li
            class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-2 fi-side-navigation-item--child-selected"
          >
            <span
              class="c2 fi-side-navigation-item_content-wrapper"
            >
              <a
                aria-current="location"
                class="c8 fi-link--router c9"
                href="/"
              >
                Crisis situations in personal finances
              </a>
              <svg
                aria-hidden="true"
                class="fi-icon c10"
                focusable="false"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="fi-icon-base-fill"
                  d="M12.99 8.39A1.433 1.433 0 0012 8c-.358 0-.717.13-.99.39l-5.6 5.334a1.287 1.287 0 000 1.885 1.448 1.448 0 001.98 0l4.61-4.39 4.61 4.39a1.448 1.448 0 001.98 0 1.287 1.287 0 000-1.885l-5.6-5.333z"
                  fill="#222"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <ul
              class="c5 fi-side-navigation-item_sub-list"
            >
              <li
                class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-3"
              >
                <span
                  class="c2 fi-side-navigation-item_content-wrapper"
                >
                  <a
                    class="c8 fi-link--router c9"
                    href="/"
                  >
                    If you are unable to pay your debts
                  </a>
                </span>
              </li>
              <li
                class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-3 fi-side-navigation-item--selected"
              >
                <span
                  class="c2 fi-side-navigation-item_content-wrapper"
                >
                  <a
                    aria-current="page"
                    class="c8 fi-link--router c9"
                    href="/"
                  >
                    Advice on banking and insurance matters
                  </a>
                </span>
              </li>
            </ul>
          </li>
          <li
            class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-2"
          >
            <span
              class="c2 fi-side-navigation-item_content-wrapper"
            >
              <a
                class="c8 fi-link--router c9"
                href="/"
              >
                Last will and testament
              </a>
            </span>
          </li>
        </ul>
      </li>
      <li
        class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-1"
      >
        <span
          class="c2 fi-side-navigation-item_content-wrapper"
        >
          <a
            class="c8 fi-link c11 fi-link--router c9 fi-link--external"
            href="https://suomi.fi"
            rel="noopener"
            target="_blank"
          >
            Taxation and public economy
            <span
              class="c2 c12 fi-visually-hidden"
            />
          </a>
        </span>
      </li>
      <li
        class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-1"
      >
        <span
          class="c2 fi-side-navigation-item_content-wrapper"
        >
          <button
            class="fi-link--router c9"
          >
            Consumer protection
          </button>
        </span>
      </li>
      <li
        aria-disabled="true"
        class="c6 c7 fi-side-navigation-item fi-side-navigation-item--level-1 fi-side-navigation-item--disabled"
      >
        <span
          class="c2 fi-side-navigation-item_content-wrapper"
        >
          <a
            class="c8 fi-link--router c9"
          >
            Finance
          </a>
        </span>
      </li>
    </ul>
  </nav>
</div>
`;
